src/gatsby/hooks/useClickOutside.tsx

```diff
@@ -0,0 +1,28 @@
+import { RefObject, useEffect } from 'react'
+
+const events = [`mousedown`, `touchstart`]
+
+const useClickOutside = <T extends HTMLElement = HTMLElement>(
+  ref: RefObject<T>,
+  onClickOutside: (event: Event) => void
+) => {
+  const isOutside = (element: Node | null) =>
+    !ref.current || !ref.current.contains(element)
+
+  const onClick = (event: Event) => {
+    if (isOutside(event.target as Node)) {
+      onClickOutside(event)
+    }
+  }
+
+  useEffect(() => {
+    for (const event of events) {
+      document.addEventListener(event, onClick)
+    }
+
+    return () => {
+      for (const event of events) document.removeEventListener(event, onClick)
+    }
+  })
```

Comment:
Do we need this `useEffect` to run on every render? Could we achieve the same thing if we just run it on once on mount 🤔

Reply:
Great and Thanks for pointing that out. Yeah, we don't need this to run on every render. We can have `ref` and `handler` as dependencies.